core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1388,6 +1388,8 @@ private[spark] class DAGScheduler(
                 if (!job.finished(rt.outputId)) {
                   job.finished(rt.outputId) = true
                   job.numFinished += 1
+                  taskScheduler.markPartitionCompletedInAllTaskSets(
```

Comment:
missed the comment about moving this higher up, so you only need it in one place?

Reply:
Makes sense, have moved it.